cmd/brew-test-bot.rb

```diff
@@ -989,7 +994,7 @@ def test_ci_upload(tap)
     remote = "git@github.com:BrewTestBot/homebrew-#{tap.repo}.git"
     git_tag = if pr
       "pr-#{pr}"
-    elsif (upstream_number = ENV["UPSTREAM_BUILD_NUMBER"])
+    elsif (upstream_number = ENV["UPSTREAM_BUILD_NUMBER"] || ENV["CIRCLE_BUILD_NUM"])
```

Comment:
Pull this into a separate variable and document that the variables are for Jenkins and Circle CI respectively.